### PR TITLE
feat: add a11y queries

### PR DESCRIPTION
**Pull request by Esemesek**
### Summary

Addition of accessibility queries to the API. Available queries:
- `byA11yLabel`, `byAccessibilityLabel`
- `byA11yHint`, `byAccessibilityHint`
- `byA11yStates`, `byAccessibilityStates`
- `byA11yRole`, `byAccessibilityRole`

### TODO:
- [x] - Add documentation

### Test plan

Unit tests included in this PR.

### CODE REVIEW

File: src/helpers/makeQuery.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// @flow
+import { ErrorWithStack } from './errors';
+
+const prepareErrorMessage = error =>
```
Comment (Esemesek) [inline src/helpers/makeQuery.js]: This is copied and pasted. Import this thing instead.

Comment (thymikee) [reply src/helpers/makeQuery.js]: Fixed that for you

File: src/helpers/a11yAPI.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  // $FlowFixMe
+  return !matcher.some(e => !prop.includes(e));
+}
+
+const a11yAPI = (instance: ReactTestInstance): A11yAPI =>
+  ({
+    ...makeQuery(
+      'accessibilityLabel',
+      {
+        getBy: 'getByA11yLabel',
```
Comment (Esemesek) [inline src/helpers/a11yAPI.js]: We could also replace `strings` with `Array<string>` to have more than one alias for a query.
E.g. `getByA11yLabel` and `getByAccessibilityLabel`. @thymikee wdyt?

Comment (thymikee) [reply src/helpers/a11yAPI.js]: I'm down 👍

**Pull request merged**